### PR TITLE
Hughsie/coverage dbx

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by superm1** [comment]
forgot `chmod +x` perhaps?

File: contrib/ci/coverage.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh -e
+
+# if invoked outside of CI
+if [ "$CI" != "true" ]; then
+    echo "Not running in CI"
+    exit 1
```
Comment (superm1) [inline contrib/ci/coverage.sh:6]: I think this should be `exit 0`

File: contrib/ci/coverage.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+fi
+
+gcovr -x \
+	--filter build/libfwupd \
+	--filter build/libfwupdplugin \
+	--filter build/plugins \
+	--filter build/src \
+	--exclude '^.+-test.c$' \
```
Comment (hughsie) [inline contrib/ci/coverage.sh:14]: @superm1 what's your view on self tests being in the coverage output? On one hand they're tests and not actually executed on user systems, on the other hand it's useful to know when bits of the tests are not being run.

Comment (superm1) [reply contrib/ci/coverage.sh:14]: I think they're worth having in it, but we need to split up codecov to use "components".  Then we can have rating for plugins, for tests, for libraries, for core etc.

Comment (superm1) [reply contrib/ci/coverage.sh:14]: https://docs.codecov.com/docs/components

Comment (superm1) [reply contrib/ci/coverage.sh:14]: Similarly after/if we get components set up I'm not sure we should be excluding all the rustgen generated stuff.  If we're generating structs but not using them, then it is a "coverage gap".  But maybe it should be a component on it's own.

File: contrib/ci/coverage.sh (hunk 1 above)
Comment (hughsie) [reply contrib/ci/coverage.sh:6]: Not so sure; I think if it's a CI-only script we shouldn't return success.

**Pull request merged**